init/tee/snp_attest.h

```diff
@@ -3,6 +3,103 @@
 #ifndef _SNP_ATTEST
 #define _SNP_ATTEST
 
+#include <stdint.h>
+
+#include <uuid/uuid.h>
+
+#define SEV_GUEST_DEV "/dev/sev-guest"
+
+/*
+ * Cryptographic signature (should be signed by the VCEK).
+ */
+struct signature {
+	uint8_t r[72];
+        uint8_t s[72];
+	uint8_t reserved[512-144];
+};
+
+/*
+ * Structure containing the security version numbers of each component in the
+ * Trusted Computing Base (TCB) of the SNP firmware.
+ */
+union tcb_version {
+	struct {
+                uint8_t boot_loader;
+		uint8_t tee;
+		uint8_t reserved[4];
+		uint8_t snp;
+		uint8_t microcode;
+	};
+	uint64_t raw;
+};
+
+/*
+ * An array of certificates. Consult the AMD SEV GHCB document to understand how
+ * this table should be built and parsed.
+ */
+struct cert_table {
+        struct cert_table_entry {
+                uuid_t guid;
+                uint32_t offset;
+                uint32_t len;
+        } *entry;
+};
+
+/*
+ * SNP attestation report structure. Based off of the attestation report
+ * structure described in firmware version 1.51.
```

Comment:
Is there any chance they've changed this struct again in `1.52`? I had to put a 32 bit padding before `report_data` to align properly its contents.

Reply:
Fixed in [1272e95](https://github.com/containers/libkrun/pull/108/commits/1272e956cf212f3877b3eb01b1f4d66719a7f8c7)